modules/packages/LinearAlgebra.chpl

```diff
@@ -2848,7 +2848,10 @@ module Sparse {
   }
 
   // Method returns the product of a Sparse and a Dense Matrix.
-  private proc sparseDenseMatmul(A: [], B: []) where isSparseArr(A) || !isSparseArr(B) {
+  private proc sparseDenseMatmul(A: [?ADom], B: [?BDom]) where isSparseArr(A) || !isSparseArr(B) {
+    if ADom.dim(1) != BDom.dim(0) then
+        halt("Mismatched shape in sparse-dense matrix multiplication");
```

Comment:
The tests look nice, thanks!  It'd be good to have a test of this error message as well :)

Reply:
Sure, should I have few for complex types too?

Reply:
Sounds good to me, yeah!